### DISCUSSION TITLE
Data Importer Authenticate with Firefly III doesn't work when not using localhost

### DISCUSSION THREAD

**Discussion by hls-management** [firefly-iii/firefly-iii#8546] [General]
Ive freshly installed firefly and the data importer on a headless vm running docker on my Mac (I use this as a test platform before I deploy via Ansible to a cluster of raspberry pi's).
I can access both the firefly and data importer ui via their ip:port combo in a browser. However when I submit my client it in the data importer the link is trying to redirect me to localhost.
```
Safari can't open the page "localhost/auth/authorize?
client_id=3&redirect_uri=http%3A%2F%2F192.168.64.88%3A81%2Fcallback&respons e_type=code&scope=&state=fTWdZB50GW2AftAo6ILFxdMicgWkfkwy4rphzvmA&code
_challenge=9FNccEm|00apYdR×7g|4Vi1b5ST-8b0u-
7kLM4QcMnl&code_challenge_method=S256" because Safari can't connect to the server "localhost".
```

the data importer recognises that it is not running on 'localhost' as it clearly states:
```
The callback URL for this installation is
http://192.168.64.88:81/callback
```

I think either:
1. ive missed a step somewhere in the config
2. there is a bug where the localhost url is hardcoded into the submit link

Anyone hit this issue before?

**Comment by JC5** [comment]
Localhost is not hard coded, I can tell you that 😅


https://docs.firefly-iii.org/how-to/data-importer/how-to-configure/

Set the VANITY_URL and the FIREFLY_III_URL as the instructions explain.

**Reply by hls-management** [reply]
I knew there was a good reason point 1 was me missing config. Thanks for the prompt reply.
